scripts/build-package.js

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env node
+'use strict';
+
+require('hard-rejection/register');
+const fs = require('fs');
+const path = require('path');
+const del = require('del');
+const execa = require('execa');
+const cpy = require('cpy');
+const makeDir = require('make-dir');
+
+const rootDir = path.resolve(__dirname, '..');
+const outputDir = path.resolve(rootDir, 'pkg');
+const srcDir = path.resolve(rootDir, 'src/components');
+
+// Build a single directory that is ready to be published as an npm package.
+// From that package components should be importable without any
+// additional nesting.
+//
+// The root package.json of this repo has `"private": true` so we don't
+// accidentally publish that way. Instead, we need to publish from the
+// directory that is built by this script.
+
+// Compile all src files that we want in the package ... so that excludes
+// tests and examples.
+function compileComponents() {
+  return execa.shell(
+    `babel "**/!(examples|__tests__)/*.js" --out-dir ${outputDir}`,
+    {
+      cwd: srcDir,
+      stdio: 'inherit'
+    }
+  );
+}
+
+// Copy non-src files that we want in the package.
+function copyFiles() {
+  return cpy(['LICENSE', 'CHANGELOG.md', 'README.md'], outputDir, {
+    cwd: rootDir
+  });
+}
+
+function copyCss() {
+  makeDir(path.resolve(rootDir, 'pkg/css')).then(path => {
+    return cpy(['src/css/*.css'], path, {
+      cwd: rootDir
+    });
+  });
+}
+
+function copyData() {
+  makeDir(path.resolve(rootDir, 'pkg/data')).then(path => {
```

Comment:
Typically you'll want to `return` the Promise in a situation like this. That way any errors will be passed through the Promise chain.